precompiles/xcm/src/tests.rs

```diff
@@ -408,6 +430,50 @@ mod xcm_new_interface_test {
         });
     }
 
+    #[test]
+    fn xtokens_transfer_works_for_native_asset() {
+        let weight = WeightV2::from(3_000_000_000u64, 1024);
+        let parent_destination = MultiLocation {
+            parents: 1,
+            interior: Junctions::X1(Junction::AccountId32 {
+                network: None,
+                id: [1u8; 32],
+            }),
+        };
+
+        ExtBuilder::default().build().execute_with(|| {
+            // sending native token to relay
+            precompiles()
+                .prepare_test(
+                    TestAccount::Alice,
+                    PRECOMPILE_ADDRESS,
+                    EvmDataWriter::new_with_selector(Action::XtokensTransfer)
+                        .write(Address::from(NATIVE_ADDRESS)) // zero address by convention
+                        .write(U256::from(42000u64))
+                        .write(parent_destination)
+                        .write(weight.clone())
+                        .build(),
+                )
+                .expect_no_logs()
+                .execute_returns(EvmDataWriter::new().write(true).build());
+
+            let expected_asset: MultiAsset = MultiAsset {
+                id: AssetId::Concrete(Here.into()),
+                fun: Fungibility::Fungible(42000),
+            };
+
+            let expected: crate::mock::RuntimeEvent =
+                mock::RuntimeEvent::Xtokens(XtokensEvent::TransferredMultiAssets {
+                    sender: TestAccount::Alice.into(),
+                    assets: vec![expected_asset.clone()].into(),
+                    fee: expected_asset,
+                    dest: parent_destination,
+                })
+                .into();
+            assert!(events().contains(&expected));
+        });
+    }
+
```

Comment:
One general comment about unit tests - I've noticed you added various negative checks in the code, but I don't see any coverage of those here.

Reply:
The modification that I have done is majorly about adding BoundedVec to check the that assets more than allowed assets are not added, for both pallet-xcm and xtokens this number is 2.

There were tests added to check if the BoundedVec is working fine for xtokens tests in the same file, so I skipped it, but if you recommend that it should be checked again, I'll add those tests.

Reply:
You mean there are tests for the new functions, but not for the old ones?

You also changed parsing of some parameters, it's _checked_ now.

Reply:
No, I meant there are tests for old functions as well which were modified according to new changes, but some of the tests were not repeated, for eg. checking BoundedVec blocks more than 2 assets.

I have added a few more tests now.